mdtraj/core/trajectory.py

```diff
@@ -1669,6 +1669,76 @@ def remove_solvent(self, exclude=None, inplace=False):
 
         return self.atom_slice(atom_indices, inplace = inplace)
 
+    def smooth(self, width=11, order=3, atom_indices=None, inplace=False):
+        """
+        Smoothen a trajectory using a zero-delay Buttersworth filter.
+        This code is copied from the scipy cookbook, with sytlistic improvements.
+        http://www.scipy.org/Cookbook/FiltFilt
+        
+        Parameters
+        ----------
+        width : float
+            This acts very similar to the window_len in the window smoother. In
+            the implementation, the frequency of the low-pass filter is taken to
+            be two over this width, so it's like "half the period" of the sinusiod
+            where the filter starts to kick in.
+        order : int, optional
+            The order of the filter. A small odd number is recommended. Higher
+            order filters cutoff more quickly, but have worse numerical
+            properties.
+        atom_indices : array-like, dtype=int, shape=(n_atoms)
+            List of indices of atoms to retain in the new trajectory.
+        inplace : bool, default=False
+            The return value is either ``self``, or the new trajectory,
+            depending on the value of ``inplace``.
+
+        Returns
+        -------
+        traj : md.Trajectory
+            The return value is either ``self``, or the new smoothed trajectory,
+            depending on the value of ``inplace``.
+        """
+        from scipy.signal import lfilter, lfilter_zi, filtfilt, butter
+
+        if width < 2.0 or not isinstance(width, int):
+            raise ValueError('width must be an integer greater than 1.')
+        if not atom_indices:
+            atom_indices = range(self.n_atoms)
+
+        xyz = self.xyz.copy()
+
+        for i in atom_indices:
+            for j in range(3):
+
+                signal = xyz[:, i, j]
+
+                # find nearest odd integer
+                pad = int(np.ceil((width + 1)/2)*2 - 1)
```

Comment:
do they scipy filter functions allow broadcasting, or is it really necessary to loop over the atom indices and dimensions.